extensions/amazon-aws/src/glue.tsx

```diff
@@ -0,0 +1,326 @@
+      markdown={
+        glueJobRun.Arguments
+          ? `**Arguments**\n\`\`\`json\n${JSON.stringify(glueJobRun.Arguments, null, 2)}\n\`\`\``
+          : undefined
+      }
+      metadata={
+        <List.Item.Detail.Metadata>
+          <List.Item.Detail.Metadata.Label title="Job Name" text={glueJobRun.JobName} />
+          <List.Item.Detail.Metadata.Label
+            title="Job Run State"
+            text={{ value: glueJobRun.JobRunState || "", color: glueJobRun.iconTintColor }}
+          />
+          <List.Item.Detail.Metadata.Label title="Execution Time" text={glueJobRun.ExecutionTime?.toString() + "s"} />
+          <List.Item.Detail.Metadata.Label title="Started on" text={glueJobRun.StartedOn?.toUTCString()} />
+          {glueJobRun.CompletedOn && (
+            <List.Item.Detail.Metadata.Label title="Completed On" text={glueJobRun.CompletedOn?.toUTCString()} />
+          )}
+          {glueJobRun.ErrorMessage && (
+            <List.Item.Detail.Metadata.Label
+              title="Error Message"
+              text={{ color: Color.Red, value: glueJobRun.ErrorMessage || "" }}
+            />
+          )}
+          <List.Item.Detail.Metadata.Label title="Arguments" text={JSON.stringify(glueJobRun.Arguments)} />
+          {glueJobRun.DPUSeconds && (
+            <List.Item.Detail.Metadata.Label title="DPU Seconds" text={glueJobRun.DPUSeconds?.toString() + "s"} />
+          )}
+          {glueJobRun.Attempt && (
+            <List.Item.Detail.Metadata.Label title="Attempt" text={glueJobRun.Attempt?.toString()} />
+          )}
+          <List.Item.Detail.Metadata.Separator />
+          <List.Item.Detail.Metadata.Label title="Worker Type" text={glueJobRun.WorkerType} />
+          {glueJobRun.WorkerType && (
+            <List.Item.Detail.Metadata.Label title="Number of Workers" text={glueJobRun.NumberOfWorkers?.toString()} />
+          )}
+          {glueJobRun.Timeout && (
+            <List.Item.Detail.Metadata.Label title="Timeout" text={glueJobRun.Timeout?.toString() + "s"} />
+          )}
+          {glueJobRun.ProfileName && (
+            <List.Item.Detail.Metadata.Label title="Profile Name" text={glueJobRun.ProfileName} />
+          )}
+          <List.Item.Detail.Metadata.Separator />
+          <List.Item.Detail.Metadata.TagList title="Versions">
+            <List.Item.Detail.Metadata.TagList.Item text={"Glue V" + glueJobRun.GlueVersion} color={Color.Yellow} />
+          </List.Item.Detail.Metadata.TagList>
+          <List.Item.Detail.Metadata.Label title="Log Group Name" text={glueJobRun.LogGroupName} />
+          <List.Item.Detail.Metadata.Label title="Id" text={glueJobRun.Id} />
+        </List.Item.Detail.Metadata>
+      }
+    />
+  );
+}
+
+async function RunGlueJob(glueJobName: string, mutate: MutatePromise<GlueJobRun[] | undefined>) {
+  mutate(new GlueClient({}).send(new StartJobRunCommand({ JobName: glueJobName })), {
+    optimisticUpdate(data) {
+      return data;
+    },
+  })
```

Comment:
This mutation is just doing a simple _revalidation_, which is not what we intend. We shouldn't be revalidating every job in case we have started a run for a particular job. So essentially, we should `optimisticUpdate` only certain fields like `StartedOn`, `JobRunState`, icon, accessoryText, etc. (whatever makes sense) and then set `shouldRevalidateAfter` to `false`.